### PR TITLE
Port Rect.scale_by(|_ip)

### PR DESCRIPTION
**Pull request by Starbuck5**
Credit goes to @dr0id for implementation and @jonotassia for keywords

This is mostly a straightforward code port, as is allowed under LGPL.

It did take me a while to get everything into the Rect/FRect system we've got, and to make sure the FRect tests demonstrated float-ness 😅

I'm not sure who wrote it, but this method has a huge test suite.

### CODE REVIEW

**Review by oddbookworm — APPROVED** [review]
LGTM!

**Review by ankith26 — COMMENTED** [review]
Even though you have correctly credited original authors in the PR comment, it would be nice to add co-authors in the git commit itself

PS: FYI I did some digging, and the tests seem to be written by the 2 people you have already credited

**Comment by Starbuck5** [comment]
I added the two original authors as co-authors on my commit.

File: docs/reST/ref/rect.rst [hunk 1]
```diff
@@ -138,6 +138,34 @@
 
       .. ## Rect.inflate_ip ##
 
+   .. method:: scale_by
+
+      | :sl:`scale the rectangle by given a multiplier`
+      | :sg:`scale_by(scale_by) -> Rect`
+      | :sg:`scale_by(scalex, scaley) -> Rect`
```
Comment (ankith26) [inline docs/reST/ref/rect.rst:145]: minor nit: I just noticed, this should be `x`/`y` not `scalex`/`scaley`

**Review by ankith26 — APPROVED** [review]
Approving this PR as it's just a port of an upstream one. The minor nitpick I mentioned can also be left for later

File: docs/reST/ref/rect.rst (hunk 1 above)
Comment (Starbuck5) [reply docs/reST/ref/rect.rst]: Good catch, updated.

**Pull request merged**